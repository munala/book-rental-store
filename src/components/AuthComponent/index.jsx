import React, { useState } from "react";
import PropTypes from "prop-types";
import Form from "react-bootstrap/Form";
import AppModal from "../common/AppModal";
import Button from "react-bootstrap/Button";

const AuthComponent = ({ loginMode, message, setMessage, onSubmit }) => {
  const pageText = loginMode ? "Login" : "Register";
  const optionButtonText = loginMode ? "Register" : "Login";
  const optionRoute = loginMode ? "register" : "login";

  const [email, setEmail] = useState("");
  const [customerId, setCustomerId] = useState("");
  const [errors, setErrors] = useState({});

  const modalButtons = [
    {
      action: () => setMessage(""),
      text: "OK"
    }
  ];

  const validateForm = () => {
    const errorObject = {};

    if (!email) {
      errorObject.email = "Email is required";
    } else {
      delete errorObject.email;
    }

    if (loginMode && !customerId) {
      errorObject.customerId = "Customer ID is required";
    } else {
      delete errorObject.customerId;
    }

    if (Object.keys(errorObject).length) {
      setErrors(errorObject);
    }

    return Object.keys(errorObject).length === 0;
  };

  const handleSubmit = event => {
    event.preventDefault();

    if (validateForm()) {
      onSubmit({
        email,
        id: customerId
      });
    }
  };

  return (
    <div id="login">
      <h3 className="text-center text-white pt-5">Book Rental Store</h3>
      <div className="container">
        <div
          id="login-row"
          className="row justify-content-center align-items-center"
        >
          <div id="login-column" className="col-md-6">
            <div id="login-box" className="col-md-12">
              <Form>
                <h3 className="text-center text-info">{pageText}</h3>
                <Form.Group>
                  <Form.Label htmlFor="email" className="text-info">
                    Email:
                  </Form.Label>
                  <Form.Control
                    type="email"
                    value={email}
                    placeholder="Enter email"
                    onChange={({ target: { value } }) => setEmail(value)}
                    isInvalid={!!errors.email}
                  />
                  <Form.Control.Feedback type="invalid">
                    {errors.email}
                  </Form.Control.Feedback>
                </Form.Group>

                {loginMode && (
                  <Form.Group>
                    <Form.Label htmlFor="customer-id" className="text-info">
                      Customer ID:
                    </Form.Label>
                    <Form.Control
                      type="password"
                      value={customerId}
                      placeholder="Enter Customer ID"
                      onChange={({ target: { value } }) => setCustomerId(value)}
                      isInvalid={!!errors.customerId}
                    />
                    <Form.Control.Feedback type="invalid">
                      {errors.customerId}
                    </Form.Control.Feedback>
                  </Form.Group>
                )}

                <div className="bottom-row-auth">
                  <Button onClick={handleSubmit} className="login-button">
                    {pageText}
                  </Button>

                  <a href={`/${optionRoute}`} className="text-info">
                    {optionButtonText}
                  </a>
                </div>
              </Form>
            </div>
          </div>
        </div>
      </div>
      <AppModal
        show={!!message}
        message={message}
        modalButtons={modalButtons}
        onClose={() => setMessage(null)}
        title="Success"
      />
    </div>
  );
};

AuthComponent.propTypes = {
  loginMode: PropTypes.bool,
  message: PropTypes.string,
  setMessage: PropTypes.func,
  onSubmit: PropTypes.func
};

export default AuthComponent;
