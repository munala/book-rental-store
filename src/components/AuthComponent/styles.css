#login .container #login-row #login-column #login-box {
  margin-top: 120px;
  max-width: 600px;
  border: 1px solid #9c9c9c;
  background-color: #eaeaea;
  padding: 20px;
}

#login-box {
  border-radius: 20px;
}

#login .container #login-row #login-column #login-box {
  padding: 20px;
}

#login .container #login-row #login-column #login-box {
  margin-top: 85px;
}

.bottom-row-auth {
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.login-button {
  background-color: #7d6959 !important;
  color: #fff;
  border-color: #7d6959 !important;
}

.login-button:hover {
  color: #ddd;
}

.text-info {
  color: #7d6959 !important;
}
