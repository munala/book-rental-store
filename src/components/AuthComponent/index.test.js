import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import AuthComponent from "./";

describe("AuthComponent tests", () => {
  test("should render", () => {
    const loginMode = true;
    const message = "";
    const setMessage = jest.fn();
    const onSubmit = jest.fn();

    const tree = renderer
      .create(
        <MemoryRouter initialEntries={["/login"]}>
          <AuthComponent
            loginMode={loginMode}
            message={message}
            setMessage={setMessage}
            onSubmit={onSubmit}
          />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
