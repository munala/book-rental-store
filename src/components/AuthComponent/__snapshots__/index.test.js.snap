// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthComponent tests should render 1`] = `
<div
  id="login"
>
  <h3
    className="text-center text-white pt-5"
  >
    Book Rental Store
  </h3>
  <div
    className="container"
  >
    <div
      className="row justify-content-center align-items-center"
      id="login-row"
    >
      <div
        className="col-md-6"
        id="login-column"
      >
        <div
          id="login-box"
        >
          <form
            className=""
          >
            <h3
              className="text-center text-info"
            >
              Login
            </h3>
            <div
              className="form-group"
            >
              <label
                className="text-info form-label"
                htmlFor="email"
              >
                Email:
              </label>
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="Enter email"
                type="email"
                value=""
              />
              <div
                className="invalid-feedback"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                className="text-info form-label"
                htmlFor="customer-id"
              >
                Customer ID:
              </label>
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="Enter Customer ID"
                type="password"
                value=""
              />
              <div
                className="invalid-feedback"
              />
            </div>
            <div
              className="bottom-row-auth"
            >
              <button
                className="login-button btn btn-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Login
              </button>
              <a
                className="text-info"
                href="/register"
                onClick={[Function]}
              >
                Register
              </a>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
